active/0000-building-target-assets.md

```diff
@@ -0,0 +1,187 @@
+
+- `preProcessTree(type, tree)`: The existing type for browser build is represented by `javascript`, `styles` etc. We need to expose additional types when targetted builds are run. Therefore, for a target build the type will be invoked as `javascript-<targetname>`, `styles-<targetname>` etc.
+- `postprocessTree(type, tree)`: The existing type for browser build is represented by `javascript`, `styles` etc. We need to expose additional types when targetted builds are run. Therefore, for a target build the type will be invoked as `javascript-<targetname>`, `styles-<targetname>` etc.
+- `lintTree(treeType, tree)`: We also want to make sure we lint the new target trees as well. Therefore we need to generate additional tree types based on the given target. The additional treeType will look as `app-<targetName>`, `templates-<targetname>` and `addon-<targetname>`.
+- `treeFor(name)`: We want to expose the similar `treeFor` hooks that are available for app, addon, templates and styles only. Other names will not apply for target builds. Some examples include vendor, public etc. Therefore additional names and their corresponding public function will be available. For example a target named as `fastboot` will invoke the additional names with `treeFor`: `app-fastboot`, `addon-fastboot`, `styles-fastboot`, `templates-fastboot`, `addon-templates-fastboot` and `addon-fastboot`. The corresponding treeFor public trees will be: `treeForFastbootApp`, `treeForFastbootAddon` etc.
+
+
+## app.import
+`app.import` allows libraries present in vendor directory or other places to be imported in the vendor file. A target may want to import a platform specific library in its vendor file. In order to allow that we need to expand the API of `app.import()` to take into account importing into target environments.
+
+Currently `app.import` takes an optional object that allows to provide additional options (for example, whether to prepend). The `options` object will now take a `targets` key which will be an array of target names where this asset needs to be imported. It will default to `browser` target if not provided.
+
+_Note_: This is optional item for this RFC.
+
+## config environments
+By default ember exposes an environment file which contains configuration in different environments (development, test and production). If a target wants to override a configuration in its environment, it should be allowed to do so. Therefore, this RFC proposes an additional directory structure
+```
+config/
+  |-- environment.js
+  |-- <targetname>
+    |-- environment.js
+```
+
+The `targetname` directory will only exists if the target environment wants to override. For the target build if the target environment file is present we will generate a new config and append it to `vendor-<targetname>.js` file, otherwise it will default to the browser environment config present in `vendor.js`.
+
+_Note_: This is optional item for this RFC.
+
+# How We Teach This
```

Comment:
It is supposed to be "Teach". It is present in the RFC template. I didn't add that section title.